### ISSUE TITLE
Support for no html pages

### ISSUE THREAD

**Issue by lazaronixon** [turbolinks/turbolinks#99]
I have pages .xhtml and turbolinks doesnt work because this line:
https://github.com/turbolinks/turbolinks/blob/ab019670632b719b3d2ecdccb158ff9f4f47746f/src/turbolinks/location.coffee#L36

Any solution or monkey path? I want use it with java and i cant change extensions.

**Comment by nateberkopec** [comment]
Checking the file extension to get the content type is weird in a web context where extensions mean little, though I can see why Turbolinks is doing it that way. I wish there was a way to compare this against `Content-Type` headers but I'm not sure that would work here.

I also recall back in the day people using the `.htm` extension, no "l".

**Comment by javan** [comment]
Perhaps we could relax the condition to `not extension? or /htm/i.test(extension)`

**Comment by lazaronixon** [comment]
I tried to ignore this condition, but links with # on the end failed.To me a propertie like Turbolinks.extension = 'xhtml', is enouth.

**Comment by packagethief** [comment]
Ultimately I think Location's `isHTML` should be configurable, as discussed in #15. Until it is (PRs welcome!) you can override `Turbolinks.Location.prototype.isHTML` after you load Turbolinks:

```js
Turbolinks.Location.prototype.isHTML = function() {
  var extension = this.getExtension()
  return extension == null || extension.match(/^\.x?html/)
}
```

**Comment by sstephenson** [comment]
We still want to make this configurable by the application in the future, but for now I think it’s reasonable to treat .htm and .xhtml extensions as being HTML. Fixed in c23dbe1.
